core/storage/header_accessor.rs

```diff
@@ -33,43 +34,58 @@ const HEADER_OFFSET_VERSION_VALID_FOR: usize = 92;
 const HEADER_OFFSET_VERSION_NUMBER: usize = 96;
 
 // Helper to get a read-only reference to the header page.
-fn get_header_page(pager: &Pager) -> Result<PageRef> {
+fn get_header_page(pager: &Pager) -> Result<CursorResult<PageRef>> {
     if pager.is_empty.load(Ordering::SeqCst) < 2 {
         return Err(LimboError::InternalError(
             "Database is empty, header does not exist - page 1 should've been allocated before this".to_string(),
         ));
     }
     let page = pager.read_page(DATABASE_HEADER_PAGE_ID)?;
-    while !page.is_loaded() || page.is_locked() {
-        // FIXME: LETS STOP DOING THESE SYNCHRONOUS IO HACKS
-        pager.io.run_once()?;
+    if !page.is_loaded() || page.is_locked() {
```

Comment:
this can actually just be `if page.is_locked()`, i wrote it wrong originally

Reply:
addressed by 2f0ff89e286b0a6499bef61e47800cd462170d07